app/src/main/java/com/example/bottomnav/StructDecoder.java

```diff
@@ -0,0 +1,59 @@
+package com.example.bottomnav;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class StructDecoder implements DataDecoder {
+    private ArrayList<StructContents> contents;
```

Comment:
Would it make sense to merge StructContents into here? Because StructContents doesn't really have a reason to exist independently from this right?

Similarly with PayloadDataType, StructContents is kind of an intermediate format, and (if the code can actual work out) I'd prefer to encapsulate the decoding logic in here. We can discuss details about how to implement this.